src/reynir_correct/pattern.py

```diff
@@ -1413,6 +1736,174 @@ def create_patterns(cls) -> None:
+                    cls.wrong_preposition_grin_af,
+                    None,
+                )
+            )
+
+            cls.add_pattern(
+                (
+                    "grín",  # Trigger lemma for this pattern
+                    "VP > { PP > { NP > { 'grín' } } PP > { 'af' } }",
+                    cls.wrong_preposition_grin_af,
+                    None,
+                )
+            )
+
+            # Catch "Hann leiðir ekki líkur af því."
+            cls.add_pattern(
+                (
+                    "leiða",  # Trigger lemma for this pattern
+                    "VP > { VP > { 'leiða' } NP-PRD > { ('líkur' | 'rök' | 'rak') } PP > { P > { 'af' } } } ",
+                    cls.wrong_preposition_leida_af,
+                    None,
+                )
+            )
+            # Catch "Hann leiðir líkur af því.", "Hann leiðir rök af því.", "Hann hefur aldrei leitt líkur af því."
+            cls.add_pattern(
+                (
+                    "leiða",  # Trigger lemma for this pattern
+                    "VP > { VP >> { VP > { 'leiða' } NP > { ('líkur' | 'rök' | 'rak') } } PP > { 'af' } } ",
+                    cls.wrong_preposition_leida_af,
+                    None,
+                )
+            )
+            # Catch "Tíminn markar upphaf af því."
+            cls.add_pattern(
+                (
+                    "upphaf",  # Trigger lemma for this pattern
+                    "VP > { VP > { 'marka' } NP-OBJ > { 'upphaf' PP > { 'af' } } }",
+                    cls.wrong_preposition_marka_af,
+                    None,
+                )
+            )
+            # Catch "Það markar ekki upphaf af því."
+            cls.add_pattern(
+                (
+                    "(upphafinn|upphaf)",  # Trigger lemma for this pattern
+                    "VP > { VP > { 'marka' } NP > { 'upphafinn' } PP > { 'af' } }",
+                    cls.wrong_preposition_marka_af,
+                    None,
+                )
+            )
+            # Catch "Það markar upphaf af því."
+            cls.add_pattern(
+                (
+                    "upphaf",  # Trigger lemma for this pattern
+                    "VP > { VP > { VP > { 'marka' } NP-SUBJ > { 'upphaf' } } PP > { 'af' } }",
+                    cls.wrong_preposition_marka_af,
+                    None,
+                )
+            )
+            # Catch "Það hefur ekki markað upphafið af því."
+            cls.add_pattern(
+                (
+                    "(upphefja|upphaf)",  # Trigger lemma for this pattern
+                    "VP > { VP > { NP > { 'markaður' } VP > { 'upphefja' } } PP > { 'af' } }",
+                    cls.wrong_preposition_marka_af,
+                    None,
+                )
+            )
+
+            # Catch "Jón leggur hann (ekki) af velli."
+            cls.add_pattern(
+                (
+                    "völlur",  # Trigger lemma for this pattern
```

Comment:
We probably need more trigger lemmas here, such as 'vell' and 'velli', thanks to BÍN.